src/main/java/com/cloudbees/hudson/plugins/folder/Folder.java

```diff
@@ -803,6 +804,16 @@ public boolean isAllowedChildDescriptor(TopLevelItemDescriptor tid) {
                 return false;
             }
         }
+        // TODO Jenkins 1.607+ use method directly
+        try {
+            return Boolean.TRUE.equals(tid.getClass().getMethod("isApplicableIn", ItemGroup.class).invoke(tid, this));
+        } catch (NoSuchMethodException e) {
+            // ignore
+        } catch (InvocationTargetException e) {
+            // ignore
+        } catch (IllegalAccessException e) {
+            // ignore
```

Comment:
The second two exceptions would be genuine problems that should be rethrown as `AssertionError`, not ignored.